### PR TITLE
Write Ignition file as a volume in a libvirt storage pool

### PR DESCRIPTION
**Pull request by eamonnotoole**
This avoids the problem where the Ignition file is remote to
the host on which the libvirt domain is being created.  We've
added a "libvirt_ignition" resource which manages the Ignition
file in the libvirt volume - creates it, and destroys it.  The
"coreos_ignition" field in the libvirt_domain definition must
point to the Id of a "libvirt_ignition" resource.

The code is modelled on that used for CloudInit.

### CODE REVIEW

File: libvirt/resource_libvirt_domain.go [hunk 1]
```diff
@@ -12,6 +9,8 @@ import (
 	"strings"
 	"time"
 
+	"crypto/sha256"
+	"encoding/hex"
```
Comment (inercia) [inline libvirt/resource_libvirt_domain.go]: Please reformat these imports.

File: libvirt/resource_libvirt_coreos_ignition.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
+package libvirt
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"log"
+)
```
Comment (eamonnotoole) [inline libvirt/resource_libvirt_coreos_ignition.go]: Done

File: libvirt/resource_libvirt_domain.go (hunk 1 above)
Comment (eamonnotoole) [inline libvirt/resource_libvirt_domain.go]: Done

**Comment by eamonnotoole** [comment]
@inercia @dmacvicar @flavio What's the prognosis with this PR?  There haven't been any comments on it in two months.

**Comment by flavio** [comment]
LGTM too

**Pull request merged**